products/surveys/backend/max_tools.py

```diff
@@ -33,33 +54,22 @@ async def _create_survey_from_instructions(self, instructions: str) -> SurveyCre
         """
         Create a survey from natural language instructions.
         """
-        # Create the prompt
-        prompt = ChatPromptTemplate.from_messages(
-            [
-                ("system", SURVEY_CREATION_SYSTEM_PROMPT),
-                ("human", "Create a survey based on these instructions: {{{instructions}}}"),
-            ],
-            template_format="mustache",
-        )
 
-        # Set up the LLM with structured output
-        model = (
-            ChatOpenAI(model="gpt-4.1-mini", temperature=0.2)
-            .with_structured_output(SurveyCreationSchema, include_raw=False)
-            .with_retry()
-        )
+        graph = FeatureFlagLookupGraph(team=self._team, user=self._user)
 
-        # Generate the survey configuration
-        chain = prompt | model
-        result = await chain.ainvoke(
-            {
-                "instructions": instructions,
-                "existing_surveys": await self._get_existing_surveys_summary(),
-                "team_survey_config": self._get_team_survey_config(self._team),
-            }
-        )
+        graph_context = {
+            "change": f"Create a survey based on these instructions: {instructions}",
+            "output": None,
+            "tool_progress_messages": [],
+            **self.context,
+        }
+
+        result = await graph.compile_full_graph().ainvoke(graph_context)
 
-        return cast(SurveyCreationSchema, result)
+        if isinstance(result["output"], SurveyCreationSchema):
+            return result["output"]
+        else:
+            return SurveyCreationSchema(questions=[], should_launch=False)
```

Comment:
i missed this on my original PR, but maybe makes sense to add an exception here? something like

```python
posthog.capture_exception(
          Exception(f"Survey creation graph returned unexpected output type: {type(result.get('output'))}"),
          {"team_id": self._team.id, "result": str(result)}
      )
```